artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ScaleDownHandler.java

```diff
@@ -578,7 +578,7 @@ public boolean lookup(MessageReference reference) throws Exception {
                   }
 
                   if (initialRef == null) {
-                     lastRef = initialRef;
+                     initialRef = lastRef;
```

Comment:
This looks like it could change behaviour and if it is indeed a bug it's obviously not covered by our test cases.  I don't know enough about this area of the code to merge this without spending significant time figuring out how it works.  @jbertram I believe you wrote this.  Could you take a look and merge if you're happy?  Cheers.